### PR TITLE
[post] 在32位 Ubuntu 16.04.1 LTS 上安装饥荒服务器

### PR DESCRIPTION
**Pull request by gaocegege**
## Post info

- Title: 在32位 Ubuntu 16.04.1 LTS 上安装饥荒服务器
- Author: gaocegege
- License: default
- Estimate finish time: done

Signed-off-by: Ce Gao <ce.gao@outlook.com>

### CODE REVIEW

**Comment by gaocegege** [comment]
PTAL @arrowrowe @at15

File: docs/在32位_ubuntu_16.04.1_lts_上安装饥荒服务器.html [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+
+<pre><code class="language-text">Internet(Upload) = 8Kbytes/player/s
+Ram = around 65Mbytes/player
+CPU = N/A
+VCRedist_2008 (x86)
+</code></pre>
+
+<p>因此选定配置的时候要计算下，服务器的最低配置要求。因为考虑到我们的玩家数最多也就20人左右，长期在线人数能在3-4人就不错了，因此一台1核2G内存的机器就可以满足我们的要求了。</p>
+
+<p>我们中的绝大多数玩家，都是在华东地区的，而只有一个美帝玩家。因此在服务器的选择上，华东节点是最合适的。在考察了包括阿里云、美团云、青云、腾讯云、Hyper.sh 在内的众多云服务提供商后，选择了最便宜的腾讯云。就流量来说，基本所有的服务商都是一个价钱，但是服务器的价格从 85 到 125 不等。Hyper.sh 因为没有华东节点，就没有关注价格。因为 steam 的 cmd 运行需要 32 位的环境，而且服务器的内存没有超过 4G，因此选择了 32位 Ubuntu 16.04.1 LTS。因为选择的云服务提供商和系统都很大众，因此在过程中并没有遇到什么坑。</p>
```
Comment (at15) [inline docs/在32位_ubuntu_16.04.1_lts_上安装饥荒服务器.html:87]: .w. just one player outside the wall /w\

File: docs/在32位_ubuntu_16.04.1_lts_上安装饥荒服务器.html [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+</code></pre>
+
+<p>然后将文件内容拷贝到 <code>/home/&lt;user&gt;/.klei/DoNotStarveTogether/Cluster_1/cluster_token.txt</code> 中就行。</p>
+
+<h2>运行</h2>
+
+<pre><code class="language-bash">/home/&lt;user&gt;/steamapps/DST/bin/dontstarve_dedicated_server_nullrenderer
+</code></pre>
+
+<p>官方推荐使用 screen 来维持服务器在退出 ssh 连接后依然在运行，但你喜欢怎么做就随便了。</p>
```
Comment (at15) [inline docs/在32位_ubuntu_16.04.1_lts_上安装饥荒服务器.html:150]: nohup? and redirect the log to somewhere, or is it logging to file directly instead of stdout

Comment (gaocegege) [reply docs/在32位_ubuntu_16.04.1_lts_上安装饥荒服务器.html:150]: 没必要这么复杂，screen 就挺好-， -

File: source/在32位_ubuntu_16.04.1_lts_上安装饥荒服务器.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+console_enabled = true
+
+[gameplay]
+max_players = <max_players_num>
+pvp = false
+game_mode = endless
+pause_when_empty = true
+```
+
+cluster_token.txt 文件的内容需要用饥荒的客户端来生成，输入~打开游戏内置的 console，输入 `TheNet:GenerateClusterToken()`，不同系统会在不同位置生成一个 token：
```
Comment (at15) [inline source/在32位_ubuntu_16.04.1_lts_上安装饥荒服务器.md]: press `~` might be better (ubuntu 16.10 got broken input method, so take my comments with a translator..., sorry about that)

**Comment by at15** [comment]
LGTM LGTM


[Approved with PullApprove](https://pullapprove.com/dyweb/blog/pull-request/5/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=dyweb/blog)

**Comment by gaocegege** [comment]
LGTM

[Approved with PullApprove](https://pullapprove.com/dyweb/blog/pull-request/5/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=dyweb/blog)

**Comment by arrowrowe** [comment]
:sheep: ~~

[Approved with PullApprove](https://pullapprove.com/dyweb/blog/pull-request/5/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=dyweb/blog)

**Pull request merged**